File: src/main/java/com/hedvig/notificationService/service/FirebaseNotificationService.java
package com.hedvig.notificationService.service;

import com.hedvig.notificationService.entities.FirebaseToken;

import java.util.Optional;

public interface FirebaseNotificationService {

  void sendNewMessageNotification(String memberId);

  void sendReferredSuccessNotification(String memberId, String referredName, String incentiveAmount, String incentiveCurrency);

  void sendSimpleNotification(String memberId, String title, String body);

  boolean sendNotification(String memberId, String body);

  Optional<FirebaseToken> getFirebaseToken(String memberId);

  void setFirebaseToken(String memberId, String token);
}

File: src/main/java/com/hedvig/notificationService/service/FirebaseNotificationService.kt
package com.hedvig.notificationService.service

import com.hedvig.notificationService.entities.FirebaseToken
import java.util.Optional

interface FirebaseNotificationService {

    fun sendNewMessageNotification(memberId: String, messageText: String?)

    fun sendReferredSuccessNotification(
        memberId: String,
        referredName: String,
        incentiveAmount: String,
        incentiveCurrency: String
    )

    fun sendConnectDirectDebitNotification(memberId: String)

    fun sendPaymentFailedNotification(memberId: String)

    fun sendClaimPaidNotification(memberId: String)

    fun sendInsurancePolicyUpdatedNotification(memberId: String)

    fun sendInsuranceRenewedNotification(memberId: String)

    fun getFirebaseToken(memberId: String): Optional<FirebaseToken>

    fun setFirebaseToken(memberId: String, token: String)

    fun sendGenericCommunicationNotification(memberId: String, titleTextKey: String, bodyTextKey: String)

    fun sendTerminatedFailedChargesNotification(memberId: String)

    fun sendHedvigReferralsEnabledNotification(memberId: String)
}
